src/mpi/coll/src/csel.c

```diff
@@ -592,13 +592,6 @@ static csel_node_s *prune_tree(csel_node_s * root, MPIR_Comm * comm_ptr)
 {
     for (csel_node_s * node = root; node;) {
         switch (node->type) {
-            case CSEL_NODE_TYPE__OPERATOR__IS_MULTI_THREADED:
-                if (MPIR_IS_THREADED == node->u.is_multi_threaded.val)
-                    node = node->success;
-                else
-                    node = node->failure;
-                break;
-
```

Comment:
Can you add a comment for this?  This pruning is supposed to be for each communicator.  But disabling this, we are technically hurt performance for non-COMM_WORLD communicators too, although the performance difference is probably too small to notice.  It'll be good to add a comment so we don't accidentally add it back in the future.

Reply:
I added a comment. Yeah performance should not be impacted much